exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -491,259 +648,563 @@ func (e *Exchange) SubmitOrder(ctx context.Context, s *order.Submit) (*order.Sub
 	if err := s.Validate(e.GetTradingRequirements()); err != nil {
 		return nil, err
 	}
-
-	fPair, err := e.FormatExchangeCurrency(s.Pair, s.AssetType)
+	var err error
+	s.Pair, err = e.FormatExchangeCurrency(s.Pair, s.AssetType)
 	if err != nil {
 		return nil, err
 	}
-	response, err := e.PlaceOrder(ctx,
-		fPair.String(),
-		s.Price,
-		s.Amount,
-		false,
-		s.Type == order.Market,
-		s.Side.IsLong())
-	if err != nil {
-		return nil, err
+	switch s.AssetType {
+	case asset.Spot:
+		var stpMode string
+		switch s.TimeInForce {
+		case order.PostOnly:
+			stpMode = "EXPIRE_MAKER"
+		case order.GoodTillCancel:
+			stpMode = "EXPIRE_TAKER"
+		}
+		switch s.Type {
+		case order.Stop, order.StopLimit, order.TrailingStop, order.TrailingStopLimit:
+			var trackingDistance string
+			if s.Type|order.TrailingStop == order.TrailingStop {
+				trackingDistance = strconv.FormatFloat(s.TrackingValue, 'f', -1, 64)
+				if s.TrackingMode == order.Percentage {
+					trackingDistance += "%"
+				}
+			}
+
+			sOrder, err := e.CreateSmartOrder(ctx, &SmartOrderRequest{
+				Symbol:         s.Pair,
+				Type:           OrderType(s.Type),
+				Side:           s.Side,
+				AccountType:    AccountType(s.AssetType),
+				Price:          s.Price,
+				StopPrice:      s.TriggerPrice,
+				Quantity:       s.Amount,
+				Amount:         s.QuoteAmount,
+				ClientOrderID:  s.ClientOrderID,
+				TimeInForce:    TimeInForce(s.TimeInForce),
+				TrailingOffset: trackingDistance,
+			})
+			if err != nil {
+				return nil, err
+			}
+			return s.DeriveSubmitResponse(sOrder.ID)
+		case order.Market, order.Limit, order.LimitMaker:
+			response, err := e.PlaceOrder(ctx, &PlaceOrderRequest{
+				AccountType:             AccountType(s.AssetType),
+				Symbol:                  s.Pair,
+				Price:                   s.Price,
+				Quantity:                s.Amount,
+				Amount:                  s.QuoteAmount,
+				AllowBorrow:             false,
+				Type:                    OrderType(s.Type),
+				Side:                    s.Side,
+				TimeInForce:             TimeInForce(s.TimeInForce),
+				ClientOrderID:           s.ClientOrderID,
+				SelfTradePreventionMode: stpMode,
+				SlippageTolerance:       "0",
+			})
+			if err != nil {
+				return nil, err
+			}
+			return s.DeriveSubmitResponse(response.ID)
+		default:
+			return nil, fmt.Errorf("%w: %v", order.ErrUnsupportedOrderType, s.Type)
+		}
+	case asset.Futures:
+		side := "BUY"
+		if s.Side.IsShort() {
+			side = "SELL"
+		}
+		var stpMode string
+		switch s.TimeInForce {
+		case order.PostOnly:
+			stpMode = "EXPIRE_MAKER"
+		case order.GoodTillCancel:
+			stpMode = "EXPIRE_TAKER"
+		}
+		switch s.Type {
+		case order.Market, order.Limit, order.LimitMaker:
+		default:
+			return nil, fmt.Errorf("%w: %v", order.ErrUnsupportedOrderType, s.Type)
+		}
+		var positionSide order.Side
+		if s.MarginType != margin.Unset {
+			positionSide = s.Side.Position()
+		}
+		response, err := e.PlaceFuturesOrder(ctx, &FuturesOrderRequest{
```

Comment:
Has no test coverage
If it does, please stop using `require.NoError` and use `assert.NoError` for multi-case tests